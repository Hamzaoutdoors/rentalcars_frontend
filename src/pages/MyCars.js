import styled from 'styled-components';
import { useSelector } from 'react-redux';
import { Grid } from '@mui/material';
import { v4 as uuidv4 } from 'uuid';
import CarCard from '../components/cars/CarCard';

const GridContainer = styled(Grid)`
  background-color: #f6f6f6;
  padding: 2rem;
  min-height: 100vh;
  width: 100%;
  max-width: 1320px;
  margin: 0 !important;
`;

const MyCars = () => {
  const cars = useSelector((state) => state.cars.data);

  return (
    <GridContainer
      container
      spacing={{ xs: 1, md: 3 }}
      columns={{
        xs: 1, sm: 8, md: 8, lg: 12,
      }}
    >
      {cars.map((item) => (
        <Grid item key={uuidv4()} xs={2} sm={4} md={4}>
          <CarCard key={uuidv4()} item={item} />
        </Grid>
      ))}
    </GridContainer>
  );
};

export default MyCars;
