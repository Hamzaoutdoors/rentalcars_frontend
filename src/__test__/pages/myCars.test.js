import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import MyCars from '../../pages/MyCars';
import store from '../../redux/configureStore';

it('render the cars list page correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <MyCars />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
