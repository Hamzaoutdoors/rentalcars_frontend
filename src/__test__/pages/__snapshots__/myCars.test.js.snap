// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the cars list page correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-spacing-md-3 sc-gKclnd fHBPrn css-1ibwkvh-MuiGrid-root"
/>
`;
